plugin/entryBase.go

```diff
@@ -181,6 +195,33 @@ func (e *EntryBase) IsSingleton() *EntryBase {
 	return e
 }
 
+// SetMetaAttributeSchema sets the meta attribute's schema. obj is an empty struct
+// that will be marshalled into a JSON schema. SetMetaSchema will panic
+// if obj is not a struct.
+func (e *EntryBase) SetMetaAttributeSchema(obj interface{}) *EntryBase {
+	s, err := schemaOf(obj)
+	if err != nil {
+		panic(fmt.Sprintf("e.SetMetaAttributeSchema: %v", err))
+	}
+	e.metaAttributeSchema = s
+	return e
+}
+
+// SetMetadataSchema sets e#Metadata's schema. obj is an empty struct
+// that will be marshalled into a JSON schema. SetMetadataSchema will
+// panic if obj is not a struct.
+//
+// NOTE: Only use SetMetadataSchema if you're overriding e#Metadata.
```

Comment:
Can you include something about why both options exist? Probably in the commit message.